src/app/reporting/Engine.cpp

```diff
@@ -336,6 +368,63 @@ void Engine::Run()
         mCurReadHandlerIdx = (mCurReadHandlerIdx + 1) % CHIP_IM_MAX_NUM_READ_HANDLER;
         readHandler        = imEngine->mReadHandlers + mCurReadHandlerIdx;
     }
+    for (auto & path : mChangeSetPathList)
+    {
+        path.ClearDirty();
+    }
+}
+
+void Engine::SetDirty(ClusterInfo & aClusterInfo)
+{
+    for (auto & path : mChangeSetPathList)
+    {
+        if (path.IsDirty())
+        {
+            //If aClusterInfo is part of existing dirty path, skip it
+            //If aClusterInfo is super set of one existing path, update that path with new path.
+            if (path.IsAttributePathSupersetOf(aClusterInfo)) {
+                return;
```

Comment:
Consider the following case:
- A path is marked dirty
- One new subscription comes
- (Report Engine run)